modules/_labs/color-picker/react/spec/ColorPicker.spec.tsx

```diff
@@ -0,0 +1,170 @@
+
+describe('<ColorPicker /> component', () => {
+  const dataTestId = 'TESTY_TEST';
+
+  const renderColorPicker = (props?: Partial<ColorPickerProps>) => {
+    return render(<ColorPicker data-test-id={dataTestId} onChange={jest.fn()} {...props} />);
+  };
+
+  it('should include aria-label', () => {
+    const ariaLabel = 'Background Color';
+    const {container, unmount} = renderColorPicker({ariaLabel});
+
+    expect(container.querySelector(`[aria-label="${ariaLabel}"]`)).not.toBeNull();
+    unmount();
+  });
+
+  it('should call onChange handler when clicking a color swatch', () => {
+    const onChange = jest.fn();
+    const {container, unmount} = renderColorPicker({onChange});
+    const colorSwatch = container.querySelector('#canvas-color-picker--color-333333')!;
+
+    fireEvent.click(colorSwatch);
+    expect(onChange).toHaveBeenCalledWith(colors.blackPepper400);
+    unmount();
+  });
+
+  it('should call onChange handler when selecting a color swatch with ENTER', () => {
+    const onChange = jest.fn();
+    const {container, unmount} = renderColorPicker({onChange});
+    const colorSwatch = container.querySelector('#canvas-color-picker--color-333333')!;
+
+    fireEvent.mouseEnter(colorSwatch);
+    fireEvent.keyDown(colorSwatch, {key: 'Enter'});
+    expect(onChange).toHaveBeenCalledWith(colors.blackPepper400);
+    unmount();
+  });
+
+  it('should display custom hex codes in the custom input', () => {
+    const {container, unmount} = renderColorPicker({
+      selectedColor: '#abcdef',
+      showCustomInput: true,
+    });
+    const input = container.querySelector('input')!;
+
+    expect(input.getAttribute('value')).toBe('abcdef');
+    unmount();
+  });
+
+  it('should NOT display known colors in the custom input', () => {
+    const {container, unmount} = renderColorPicker({
+      selectedColor: colors.blueberry400,
+      showCustomInput: true,
+    });
+    const input = container.querySelector('input')!;
+
+    expect(input.getAttribute('value')).toBe('');
+    unmount();
+  });
+
+  describe('custom hex input', () => {
+    it('should NOT render by default', () => {
+      const {container, unmount} = renderColorPicker();
+
+      expect(container.querySelector('input')).toBeNull();
+      unmount();
+    });
+
+    it('should render when prop is true', () => {
+      const {container, unmount} = renderColorPicker({showCustomInput: true});
+
+      expect(container.querySelector('input')).not.toBeNull();
+      unmount();
+    });
+
+    it('should call onChange handler when entering custom hex code', () => {
+      const onChange = jest.fn();
+      const {container, unmount} = renderColorPicker({onChange, showCustomInput: true});
+      const input = container.querySelector('input')!;
+
+      fireEvent.mouseEnter(input);
+      fireEvent.change(input, {target: {value: '123456'}});
+      fireEvent.keyDown(input, {key: 'Enter'});
+      expect(onChange).toHaveBeenCalledWith('#123456');
+      unmount();
+    });
```

Comment:
This whole test belongs in Cypress. These types of tests are difficult to write, debug and maintain in React Testing Library